### PR TITLE
fix: fixed hash arg sorting in merkle tree generation

### PR DESCRIPTION
**Pull request by clexmond**
## Motivation and Resolution
Merkle tree generation was relying on a sorted pedersen hash function (as described in #341 ). The sorting function was doing a string compare which can easily fail when comparing to hex strings. For example: `'0xa' > '0x12' = true` when of course `10 > 18 = false`. Given that the vast majority of (all?) on-chain Merkle tree verifiers also sort this causes mismatching roots and proofs with higher and higher likelihood the larger the number of leaves.

This change simply casts the hex strings to BN to compare.

### RPC version (if applicable)

N/A

## Usage related changes

No usage / API changes.

## Development related changes

No development related changes.

## Checklist:

- [x] Performed a self-review of the code
- [x] Rebased to the last commit of the target branch (or merged it into my branch)
- [x] Linked the issues which this PR resolves
- [x] Documented the changes
- [x] Updated the docs (www)
- [x] Updated the tests
- [x] All tests are passing

### CODE REVIEW

File: src/utils/merkle.ts [hunk 1]
```diff
@@ -31,7 +32,14 @@ export class MerkleTree {
...
   static hash(a: string, b: string) {
-    const [aSorted, bSorted] = [a, b].sort();
+    let aSorted = a;
+    let bSorted = b;
+
+    if (toBN(aSorted) > toBN(bSorted)) {
+      aSorted = b;
+      bSorted = a;
+    }
+
```
Comment (tabaktoni) [inline src/utils/merkle.ts:42]: Nitpick suggestions
```suggestion
const [aSorted, bSorted] = [toBN(a), toBN(b)].sort((x: any, y: any) => x - y);
```

or in one line

```
  static hash = (a: string, b: string) =>
    pedersen(toBN(a) < toBN(b) ? [toBN(a), toBN(b)] : [toBN(b), toBN(a)]);
```

**Review by tabaktoni — APPROVED** [review]
Nice catch! also with dev docs.
Indead functionless sort do it by string order not number

File: src/utils/merkle.ts (hunk 1 above)
Comment (clexmond) [reply src/utils/merkle.ts]: I think the first one is good, I adapted it slightly to ensure we avoid any type / casting weirdness. Also realized that even the `>` with BNs had failing edge cases so I fixed and added an additional test case.

**Review by ivpavici — APPROVED** [review]
